dulwich/objectspec.py

```diff
@@ -54,6 +54,8 @@ def parse_tree(repo, treeish):
       KeyError: If the object can not be found
     """
     treeish = to_bytes(treeish)
+    if treeish not in repo.object_store.__iter__():
```

Comment:
Please don't use __iter__ but __contains__ - __iter__ can be very slow on large object stores. Why is the code on line 59 not sufficient? Tha twould have the same ffect.

Reply:
I would like to match treeish in the `repo.refs` while treeish is not a **commit sha** . It seem it isn't a good idea. Do you have better solution? what about replace `repo.object_store.__iter__()` with `list(repo.object_store)`?

Reply:
You should be able to rely on \_\_contains\_\_, or better yet - rely on just accessing the treeish and handling the exception if that doesn't work.

\_\_iter\_\_ and list() both traverse all the objects in the object store, which can be millions of objects for larger projects like the linux kernel.

Reply:
Thanks, updated in the new commit